endaq/ide/nmea.py

```diff
@@ -0,0 +1,179 @@
+from pynmeagps import NMEAReader, NMEAMessage
+from idelib.dataset import Dataset
+import pandas as pd
+import datetime as dt
+import endaq.ide.measurement as measure
+# import os.path
+# import endaq.ide.files  # used in commented-out testing at bottom of file
+
+
+def get_nmea(buffer: bytearray, raw=False):
+    """ Simple NMEA sentence extractor.
+
+        :param buffer: a bytearray containing bytes from an IDE dataset
+        :param raw: if true, return raw NMEA bytes as opposed to parsed objects. Defaults to false
+        :returns: a list of extracted sentences and the leftover buffer (which could have partial sentences completed
+            in later blocks).
+        :raise ValueError: if buffer is not a bytearray
+    """
+    if not isinstance(buffer, bytearray):
+        raise ValueError(f"Bad type for buffer: {type(buffer)}")
+    sentences = []
+    while b'\r\n' in buffer:
+        chunk, _delimiter, buffer = buffer.partition(b'\r\n')
+        _junk, prefix, chunk = chunk.partition(b'$G')
+        if chunk:
+            if raw:
+                sentences.append(prefix + chunk + _delimiter)
+            else:
+                sentences.append(NMEAReader.parse(bytes(prefix + chunk + _delimiter)))
+    return sentences, buffer
+
+
+def get_nmea_sentence(dataset: [Dataset], channel=None, raw=False):
```

Comment:
This should probably be plural (`get_nmea_sentences()`), since it returns a list of sentences.